core/iwasm/common/wasm_shared_memory.c

```diff
@@ -111,15 +111,21 @@ notify_stale_threads_on_exception(WASMModuleInstanceCommon *module_inst)
 {
     AtomicWaitAddressArgs args = { 0 };
     uint32 i = 0, total_elem_count = 0;
+    uint64 total_elem_count_size = 0;
 
     os_mutex_lock(&shared_memory_list_lock);
 
     /* count number of addresses in wait_map */
     bh_hash_map_traverse(wait_map, wait_map_address_count_callback,
                          (void *)&total_elem_count);
 
+    if (!total_elem_count)
+        return;
```

Comment:
should you unlock `shared_memory_list_lock` here?

Reply:
missed, done